completions/apt-mark

```diff
@@ -0,0 +1,52 @@
+# Debian apt-mark(8) completion                             -*- shell-script -*-
+
+_apt_mark()
+{
+    local cur prev words cword split
+    _init_completion -s || return
+
+    local special i
+    for (( i=0; i < ${#words[@]}-1; i++ )); do
+        if [[ ${words[i]} == @(hold|unhold|auto|manual|install|remove|purge|showhold|showauto|showmanual|showinstall|showremove|showpurge|minimize-manual) ]]; then
+            special=${words[i]}
+        fi
+    done
+
+    if [[ -n $special ]]; then
+        case $special in
+            minimize-manual)
+                return
+                ;;
+            *)
+                COMPREPLY=( $(apt-cache --no-generate pkgnames "$cur" \
+                    2>/dev/null) )
+                ;;
+        esac
+        return
+    fi
+
+    case $prev in
+        --help|--version|--option|-!(-*)[hvo])
+            return
+            ;;
+        --config-file|--file|-!(-*)[cf])
+            _filedir
```

Comment:
Do you thing we could add some file extensions to narrow down the completions here? `*.conf` would probably work well enough for configs (I see we don't do that for apt-get either for reasons unknown to me though). Do the stats files accepted by `--file` follow a naming pattern we could use?

Reply:
I've added `*.conf` in the latest version for `--config-file`, but for `--file=` I'm not 100% sure if there are any extensions, since by default it seems to read the file `/var/lib/apt/extended_states` which doesn't have one.